compiler/runtime/OMRSharedCacheRelocationRuntime.hpp

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_SHARED_CACHE_RELOCATION_RUNTIME_INCL
+#define OMR_SHARED_CACHE_RELOCATION_RUNTIME_INCL
+
+#ifndef OMR_SHARED_RELOCATION_RUNTIME_CONNECTOR
+#define OMR_SHARED_RELOCATION_RUNTIME_CONNECTOR
+namespace OMR { class SharedCacheRelocationRuntime; }
+namespace OMR { typedef OMR::SharedCacheRelocationRuntime SharedCacheRelocationRuntimeConnector; }
+#endif
+
+#include "runtime/RelocationRuntime.hpp"
+#include <map>
+
+namespace TR { class RelocationRuntime; }
+
+namespace OMR {
+class OMR_EXTENSIBLE SharedCacheRelocationRuntime : public TR::RelocationRuntime {
```

Comment:
I'm not sure I know why we need to have a `SharedCacheRelocationRuntime`. Couldn't everything be in `RelocationRuntime`. I don't even know why in OpenJ9 there's two relocation runtimes, we never instantiate a `TR_RelocationRuntime`, we always instantiate a `TR_SharedCacheRelocationRuntime`...

I think for OMR, just having `TR::RelocationRuntime` is sufficient.

Reply:
Hello @dsouzai , I was thinking to include shared cache relocation runtime as one that has knowledge of shared cache. Is that a bad idea? I thought there might be a need to have a relocation runtime without a Shared Cache?

Reply:
Hm I suppose that's reasonable. I'm a proponent of the idea that premature optimization is the root of all evil, and so given that all of the support here is tailored to relocation assuming a Shared Cache, anyone who wants to implement relocation without a Shared Cache would have to do a non-trivial amount of work anyway such that if they needed to split up `SharedCacheRelocationRuntime` into two parts, it wouldn't be that much extra work.

However, since you already have it split up I suppose there's no real harm leaving it as is.

Reply:
Thank you, i consider this resolved!